cognite/client/_api/datapoints.py

```diff
@@ -600,16 +603,16 @@ def __init__(self, config: ClientConfig, api_version: Optional[str], cognite_cli
     def retrieve(
         self,
         *,
-        id: Union[None, int, Dict[str, Any], Sequence[Union[int, Dict[str, Any]]]] = None,
-        external_id: Union[None, str, Dict[str, Any], Sequence[Union[str, Dict[str, Any]]]] = None,
-        start: Union[int, str, datetime, None] = None,
-        end: Union[int, str, datetime, None] = None,
-        aggregates: Union[str, List[str], None] = None,
+        id: None | int | dict[str, Any] | Sequence[int | dict[str, Any]] = None,
```

Comment:
TIL! Didn't know pep604 and pep585 were backported to 3.8.

Reply:
We can make the pyupgrade linter in ruff apply this consistently accross the code base in a separate PR. (We may have to manually add `from __future__ import annotations` import statements where they're missing)

Reply:
> TIL! Didn't know pep604 and pep585 were backported to 3.8.

Don't you remember reviewing this (`cognite/client/utils/_validation.py`) a few weeks back? 😜 

> We can make the pyupgrade linter in ruff apply this consistently across the code base in a separate PR

Although the future-import allows most of 604 and 585, there was a reason why I used the `pyupgrade` setting `--keep-runtime-typing` (to keep it from doing exactly this) as it ruined some of the `cast(<type(s)>, variable)`; these do not support... before:

```py
cast(list[str], 123)  # 3.9+
cast(str | list, 123)  # 3.10+
```

Maybe ruff is more clever and keeps these untouched?

> We may have to manually add from __future__ import annotations import statements where they're missing
There must be a code mod tool to auto-add the future-import, right 🤔  If not, back when I added all these, I wrote a quick script which we could add to pre-commit (since we already use a custom check that checks version).